src/hotspot/share/prims/whitebox.cpp

```diff
@@ -2025,6 +2025,29 @@ WB_ENTRY(jint, WB_HandshakeWalkStack(JNIEnv* env, jobject wb, jobject thread_han
   return tsc.num_threads_completed();
 WB_END
 
+WB_ENTRY(void, WB_AsyncHandshakeWalkStack(JNIEnv* env, jobject wb, jobject thread_handle))
+  class TraceSelfClosure : public AsynchHandshakeClosure {
+    void do_thread(Thread* th) {
+      assert(th->is_Java_thread(), "sanity");
+      JavaThread* jt = (JavaThread*)th;
```

Comment:
Can whitebox.cpp code use the new as_Java_thread() call?

Reply:
Fixed :)